### PR TITLE
Add zoneinfo data for System.Runtime

### PR DESCRIPTION
**Pull request by tqiu8**
Add zoneinfo data for System.Runtime tests.

### CODE REVIEW

File: src/System.Runtime.TestData/zoneinfo/mono-webassembly-zoneinfo-fs.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+          fetched = data;
+        }
+      }, handleError);
+    
+  function runWithFS() {
+
+    function assert(check, msg) {
+      if (!check) throw msg + new Error().stack;
+    }
+Module['FS_createPath']('/', 'zoneinfo', true, true);
```
Comment (safern) [inline src/System.Runtime.TestData/zoneinfo/mono-webassembly-zoneinfo-fs.js:101]: Curious, why do we need to create all these paths here as well, they're created in `runtime-test.js`

**Comment by safern** [comment]
I don't see `zoneinfo/Europe/London` -- our tests use that:

```
WASM: [FAIL] System.Tests.TimeZoneInfoTests.ConvertTime_SameTimeZones
        WASM: System.TypeInitializationException : The type initializer for 'System.Tests.TimeZoneInfoTests' threw an exception.
        WASM: ---- System.TimeZoneNotFoundException : The time zone ID 'Europe/London' was not found on the local computer.
        WASM: -------- System.IO.DirectoryNotFoundException : Could not find a part of the path '/usr/share/zoneinfo/Europe/London'.
```

File: src/System.Runtime.TestData/zoneinfo/mono-webassembly-zoneinfo-fs.js (hunk 1 above)
Comment (tqiu8) [reply src/System.Runtime.TestData/zoneinfo/mono-webassembly-zoneinfo-fs.js:101]: Oh, we actually don't need this file at all. This was the original data with wasm in mono/mono. I can go ahead and remove it.

**Comment by tqiu8** [comment]
@safern `zoneinfo/Europe/London` is in the metadata:
`{"start":651705,"audio":0,"end":655366,"filename":"/zoneinfo/Europe/London"},`

**Comment by safern** [comment]
:facepalm: — didn’t see it. Thanks.

File: src/System.Runtime.TestData/zoneinfo/mono-webassembly-zoneinfo-fs.js (hunk 1 above)
Comment (safern) [reply src/System.Runtime.TestData/zoneinfo/mono-webassembly-zoneinfo-fs.js:101]: Cool.

**Comment by safern** [comment]
Let me know if you need guidance on how to consume this new package in `dotnet/runtime`

**Comment by tqiu8** [comment]
@safern that would be great actually. It looks like adding a dependency in `eng/Versions.Details.xml` and then a 
```
<ItemGroup>
    <Content Include="$(MSBuildThisFileDirectory)TestData\**\*"
          Link="%(RecursiveDir)%(Filename)%(Extension)"
          CopyToOutputDirectory="PreserveNewest" />
  </ItemGroup>
```

**Comment by safern** [comment]
> Does that look correct?

Sort of. The `Content` piece shouldn't be needed as these packages automatically add all of it's contents to `Content` by adding a package reference and then the work I did for `Content` to be mapped into the VFS will make these files be present under `supportFiles` in the `AppDir`. 

So these two lines will need to read from `supportFiles` directory:
https://github.com/dotnet/runtime/blob/master/src/mono/wasm/runtime-test.js#L232-L233

To consume it, you need to add a dependency in `eng/Versions.Details.xml`, then add an entry in `eng/Versions.props`:

https://github.com/dotnet/runtime/blob/master/eng/Versions.props#L75

```xml
 <SystemRuntimeTestDataVersion>{packageVersion}</SystemRuntimeTestDataVersion>
```

And last you need to add a package reference in the test projects that need this data, using the `$(SystemRuntimeTestDataVersion)` property as the package version. https://github.com/dotnet/runtime/blob/ee2355c801d892f2894b0f7b14a20e6cc50e0e54/src/libraries/System.ComponentModel.TypeConverter/tests/System.ComponentModel.TypeConverter.Tests.csproj#L160

However I just looked at the official build after your change and the package is not being shipped because we missed to add it to the solution. 🤕.
https://github.com/dotnet/runtime-assets/blob/master/runtime-assets.sln

Could you put up a PR to add it and then we can get the package version?

**Comment by tqiu8** [comment]
@safern oops yes let me do that.

**Comment by safern** [comment]
@tqiu8 the generated package is: 

System.Runtime.TestData.5.0.0-beta.20319.2.nupkg

Got it from: https://dnceng.visualstudio.com/internal/_build/results?buildId=696371&view=logs&j=fa59fe4e-195c-56fa-189b-58fd241f10dd&t=afa1ce5e-d2bc-56f3-ad78-7883f25a3e3a&l=71

**Pull request merged**